providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java

```diff
@@ -0,0 +1,331 @@
+package dev.openfeature.contrib.providers.statsig;
+
+import com.statsig.sdk.DynamicConfig;
+import com.statsig.sdk.Layer;
+import com.statsig.sdk.Statsig;
+import com.statsig.sdk.StatsigUser;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.MutableContext;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Structure;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Provider implementation for Statsig.
+ */
+@Slf4j
+public class StatsigProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "Statsig";
+
+    public static final String PROVIDER_NOT_YET_INITIALIZED = "provider not yet initialized";
+    public static final String UNKNOWN_ERROR = "unknown error";
+    public static final String FEATURE_CONFIG_KEY = "feature_config";
+
+    private final StatsigProviderConfig statsigProviderConfig;
+
+    @Getter
+    private ProviderState state = ProviderState.NOT_READY;
+
+    private final AtomicBoolean isInitialized = new AtomicBoolean(false);
+
+    /**
+     * Constructor.
+     * @param statsigProviderConfig StatsigProvider Config
+     */
+    public StatsigProvider(StatsigProviderConfig statsigProviderConfig) {
+        this.statsigProviderConfig = statsigProviderConfig;
+    }
+
+    /**
+     * Initialize the provider.
+     * @param evaluationContext evaluation context
+     * @throws Exception on error
+     */
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        boolean initialized = isInitialized.getAndSet(true);
+        if (initialized) {
+            throw new GeneralError("already initialized");
+        }
+        super.initialize(evaluationContext);
+
+        Future<Void> initFuture = Statsig.initializeAsync(statsigProviderConfig.getSdkKey(),
+            statsigProviderConfig.getOptions());
+        initFuture.get();
+
+        statsigProviderConfig.postInit();
+        state = ProviderState.READY;
+        log.info("finished initializing provider, state: {}", state);
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    /**
+     * Feature config, as required for evaluation.
+     */
+    @AllArgsConstructor
+    @Getter
+    public static class FeatureConfig {
```

Comment:
I think this fits after well at the end of the file, unmixed from public methods